### PR TITLE
Allen & Heath Xone K2/K1 mapping part 2

### PR DESCRIPTION
**Pull request by Be-ing**
I realized I could fit in the `loop_in` and `loop_out` Controls for manual loops with the supershift mode. This works well because these Controls don't require showing any state on the LEDs.

### CODE REVIEW

**Comment by Be-ing** [comment]
I have added a new mode that allows for controlling the parameters of all the effects in one unit at a time. It is only available in the 4 effect unit layouts and is activated with the Layer button in the bottom left. Documentation is on the [wiki](https://mixxx.org/wiki/doku.php/allen_heath_xone_k2#effect_unit_focusing).

File: res/controllers/Xone-K2-scripts.js [hunk 1]
```diff
@@ -424,6 +427,9 @@ XoneK2.Deck = function (column, deckNumber, midiChannel) {
         shift: function () {
             this.inKey = 'beatlooproll_activate';
         },
+        supershift: function () {
+            this.inKey = 'loop_out';
+        },
```
Comment (ferranpujolcamins) [inline res/controllers/Xone-K2-scripts.js]: This is pending to be documented in the wiki, right?

**Review by ferranlala — CHANGES_REQUESTED** [review]
Your new mode is nice! However, I miss the possibility to change effects without having to exit the focused mode.

File: res/controllers/Xone-K2-scripts.js (hunk 1 above)
Comment (Be-ing) [reply res/controllers/Xone-K2-scripts.js]: yes

**Comment by Be-ing** [comment]
> I miss the possibility to change effects without having to exit the focused mode.

I thought about that but I am not sure where to fit it into the design. Maybe shift + the first parameter knob?

**Comment by ferranpujolcamins** [comment]
You could map each parameter knob

**Comment by Be-ing** [comment]
That happens to make the code a little simpler, so I'll do that.

**Comment by Be-ing** [comment]
Now that I have implemented this new mode, I am really looking forward to a GSOC student implementing parameter rearrangement this summer. It would be cool to make that somehow mappable for controller, for example maybe a new ControlObject to cycle the last parameter to the first parameter position. I'll have to think about this more...

File: res/controllers/Xone-K2-scripts.js (hunk 1 above)
Comment (Be-ing) [reply res/controllers/Xone-K2-scripts.js]: documented

**Comment by Be-ing** [comment]
Okay, I implemented effect switching by turning any parameter knob with shift in the effect unit focused mode.

I changed the 2 deck + 2 effect unit layouts so the bottom two effect routing buttons control the routing for the master and headphone channels. I don't think there's a point to having those mapped for routing to decks 3 and 4 when those decks are not mapped on the controller. Also, with the 4 effect unit layout, the deck 3 & 4 routing buttons switch to controlling the master and headphones channels with shift and turn amber instead of red.

I'm thinking about switching the shifted and unshifted functions of the bottom two amber buttons for the decks in loop mode. That is, they'd do beatjumping normally and loop double/halve with shift. What do you think about that?

**Comment by ferranpujolcamins** [comment]
> I'm thinking about switching the shifted and unshifted functions of the bottom two amber buttons for the decks in loop mode. That is, they'd do beatjumping normally and loop double/halve with shift. What do you think about that?

I don't have a strong preference about that. What you prefer will be ok :)

About your last changes, I won't be able to test them soon.

**Comment by Be-ing** [comment]
@ferranpujolcamins can you do one last check to make sure nothing is obviously messed up?

**Comment by Be-ing** [comment]
I just played a set with this for a few hours. I am quite happy with the decision to make beatjumping the default action of the bottom buttons instead of loop halve/double. It is very useful on a controller without jog wheels.

**Pull request merged**